articles/exploring-matplotlib-for-visualizations-in-python/index.md

```diff
@@ -0,0 +1,315 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: exploring-matplotlib-for-visualizations-in-python
+title: Exploring Matplotlib for Visualizations in Python
+description: Covering numpy, matplotlib - exploring matplotlib features for data visualization in python. We cover the various use cases and generate 2-D, 3-D plots and various charts for data analysis.
+author: lalithnarayan-c
+date: 2020-07-30T00:00:00-08:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/exploring-matplotlib-for-visualizations-in-python/hero.png
+    alt: matplotlib example image
+---
+Matplotlib is one of the most powerful and popular plotting libraries for Python and the numerical extension NumPy. It enables the creation of static, animated, and interactive visualizations in Python. Reading the matplotlib documentation is always ideal, but the amount of information available can be daunting. Therefore the following article is designed to gain a basic understanding of the library. Further, we will also understand the use cases of the library.
+<!--more-->
+
+### NumPy
+Numpy is the scientific computing library for Python. It is a library highly optimized written in C to perform faster vectorized, indexing, and distributed operations. It allows operations that fall under the domains of image processing, 3-D visualizations, interactive computing. astronomical processes, simulation modeling to name a few.
+
+NumPy is the library that one must know to excel in any computerized field in today's world. Let's look at some of the most commonly used functions:
+
+#### Coding Time
+
+Let's begin by creating a numpy array
+
+```python
+import numpy as np
+a=np.array([1,2,3,4,5])
+```
+The input to the np.array should be a list. To specify the dtype at runtime, we manually add the argument dtype in the np.array function.
```

Comment:
`To specify the data type at runtime, we manually add the argument dtype in the np.array function.` makes more sense to me.